src/EdjCase.JsonRpc.Router/Defaults/DefaultRpcInvoker.cs

```diff
@@ -659,12 +659,18 @@ private bool TryParseParameterList(MethodInfo method, Dictionary<string, object>
 			parameterList = new object[parameterInfoList.Count()];
 			foreach (ParameterInfo parameterInfo in parameterInfoList)
 			{
-				if (!parametersMap.ContainsKey(parameterInfo.Name) && !parameterInfo.IsOptional)
+				if (!parametersMap.ContainsKey(parameterInfo.Name))
```

Comment:
Im trying to understand what is going on here. if the parameter is not optional it should be specified. I believe this method is used to help identify method signatures that are matching. This might make more methods ambiguous

Reply:
I think this makes sense now, just need the other things fixed, or i can just pull the changes and fix them